File: enzevalos_iphone/PGPTemporaryDecryptionObject.m
//
//  PGPTemporaryDecryptionObject.m
//  enzevalos_iphone
//
//  Created by jakobsbode on 16.02.17.
//  Copyright © 2017 fu-berlin. All rights reserved.
//

//#import <Foundation/Foundation.h>
#import "PGPTemporaryDecryptionObject.h"

@implementation PGPTemporaryDecryptionObject

- (instancetype)init:(PGPSignaturePacket*)signaturePacket plaintextData:(NSData *)plaintextData
{
    self->_signaturePacket = signaturePacket;
    self->_plaintextData = plaintextData;
    if (signaturePacket != nil) {
        self->_incompleteKeyID = signaturePacket.issuerKeyID.longKeyString;
    }else {
        self->_incompleteKeyID = nil;
    }
    
    return self;
}

@end

File: enzevalos_iphone/PGPTemporaryDecryptionObject.swift
//
//  PGPTemporaryDecryptionObject.swift
//  enzevalos_iphone
//
//  Created by jakobsbode on 09.02.17.
//  Copyright © 2017 fu-berlin. All rights reserved.
//


class PGPTemporaryDecryptionObject {
    let onePassSignaturePacket: PGPOnePassSignaturePacket
    let signaturePacket: PGPSignaturePacket
    let plaintextData: Data?
    
    init(onePassSignaturePacket: PGPOnePassSignaturePacket, signaturePacket: PGPSignaturePacket, plaintextData: Data?){
        self.onePassSignaturePacket = onePassSignaturePacket
        self.signaturePacket = signaturePacket
        self.plaintextData = plaintextData
    }
    
}
